app/models/reminder_mailing_volunteer.rb

```diff
@@ -0,0 +1,32 @@
+class ReminderMailingVolunteer < ApplicationRecord
+  attr_accessor :selected
+
+  before_create :remove_or_process_email_content
+
+  belongs_to :reminder_mailing
+  belongs_to :volunteer
+  belongs_to :reminder_mailable, polymorphic: true, optional: true
+
+  private
+
+  def remove_or_process_email_content
+    return destroy if selected && selected.to_i.zero?
```

Comment:
For nested attributes / auto-save associations you can pass the magic field `_destroy` to indicate whether the record should be deleted, i.e. you can remove the `selected` attribute and this line and try just having a checkbox for `_destroy` in the view (not sure if it works as expected in this case)